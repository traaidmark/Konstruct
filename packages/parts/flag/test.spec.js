// PARTS [ FLAG > TESTS ] ######################################################

// 1. DEPENDENCIES +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// 1.1. EXTERNAL DEPENDENCIES ..................................................

import React from 'react';
import renderer from 'react-test-renderer';

// 1.1. END ....................................................................

// 1.2. INTERNAL DEPENDENCIES ..................................................

import Flag from './src';

// 1.2. END ....................................................................

// 1. END ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// 2. PROPS ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

const props = {
  text: 'usd',
}

// 2. END ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// 3. TEST +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

describe('Parts: Flag', () => {

  // 3.1. RENDER WITHOUT PROPS .................................................

  test('renders correctly without any props', () => {
    const tree = renderer.create(
      <Flag />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // 3.1. END ..................................................................

  // 3.2. RENDER WITH ALL PROPS ................................................

  test('renders correctly with all props', () => {
    const tree = renderer.create(
      <Flag { ...props } />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // 3.1. END ..................................................................

});

// 3. END ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// END OF FILE #################################################################
