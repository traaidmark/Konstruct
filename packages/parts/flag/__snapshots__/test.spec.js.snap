// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parts: Flag renders correctly with all props 1`] = `
.c0 {
  --flag-radius: var(--radius-small);
  --flag-info-background-color: var(--color-blue);
  --flag-info-text-color: var(--color-darker);
  --flag-new-background-color: var(--color-green);
  --flag-new-text-color: var(--color-darker);
  --flag-warning-background-color: var(--color-orange);
  --flag-warning-text-color: var(--color-darker);
  --flag-error-background-color: var(--color-red);
  --flag-error-text-color: var(--color-darker);
  font-size: var(--font-small-size);
  line-height: var(--font-small-lineHeight);
  font-weight: var(--weight-bold);
  background-color: var(--flag-info-background-color)!important;
  color: var(--flag-info-text-color)!important;
  padding: 2px var(--gutter-small);
  border-radius: var(--flag-radius);
}

<span
  className="c0"
>
  usd
</span>
`;

exports[`Parts: Flag renders correctly without any props 1`] = `null`;
